src/protocol/reveal.rs

```diff
@@ -36,6 +40,27 @@ pub async fn reveal<F: Field>(
     Ok(input.left() + input.right() + share)
 }
 
+/// Given a vector containing secret shares of a permutation, this returns a revealed permutation.
+/// This executes `reveal` protocol on each row of the vector and then constructs a `Permutation` object
+/// from the revealed rows.
+#[allow(clippy::cast_possible_truncation, clippy::module_name_repetitions)]
+pub async fn reveal_a_permutation<F: Field>(
+    ctx: ProtocolContext<'_, F>,
+    permutations: &mut [Replicated<F>],
+) -> Result<Permutation, BoxError> {
+    let reveals =
+        zip(repeat(ctx), permutations)
+            .enumerate()
+            .map(|(index, (ctx, input))| async move {
+                reveal(ctx, RecordId::from(index as u32), *input).await
+            });
+    let permutation = try_join_all(reveals).await?;
+    let mut perms = Vec::new();
+    for i in permutation {
+        perms.push(i.as_u128() as usize);
+    }
+    Ok(Permutation::oneline(perms))
```

Comment:
I'm pretty sure we will eventually get rid of this permutations crate if it makes us do all these extra operations and use all this extra memory.

Reply:
:(